### PR TITLE
Fix/bugs

### PR DESCRIPTION
**Pull request by WhiteNik16**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - `@rarimo/react-nft-checkout`: React implementation of `@rarimo/nft-checkout` package
 - `@rarimo/react-nft-checkout`: Exported UI component `RarimoPayButon`
 - `@rarimo/react-nft-checkout`: Exported context provider `DappContextProvider` with core methods for NFT chekout
+- `@rarimo/nft-checkout`: Added selectable token to swap
+- `@rarimo/nft-checkout`: disabled native token and wrrapped native tokens
```
Comment (napalmpapalam) [inline CHANGELOG.md:14]: remove Added
start sentence from big letter

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
+### Fixed
+- `@rarimo/nft-checkout`: Adding chain
+- `@rarimo/provider`: Adding chain
```
Comment (napalmpapalam) [inline CHANGELOG.md:18]: not clear what does it mean

File: packages/nft-checkout/src/operations/evm/index.ts [hunk 3]
```diff
@@ -148,10 +150,16 @@ export class EVMOperation
     return this.#chains
   }
 
-  public async supportedTokens(): Promise<Token[]> {
+  public async supportedTokens(chain: BridgeChain): Promise<Token[]> {
```
Comment (napalmpapalam) [inline packages/nft-checkout/src/operations/evm/index.ts:153]: make this argument optional, in other case, it should use `chainFrom`

File: packages/nft-checkout/src/operations/evm/index.ts [hunk 4]
```diff
@@ -350,7 +356,7 @@ export class EVMOperation
     try {
       await this.#provider.switchChain(targetChain!.id)
     } catch (e) {
-      if (!(e instanceof providerErrors.ProviderChainNotFoundError)) {
+      if (e instanceof providerErrors.ProviderChainNotFoundError) {
```
Comment (napalmpapalam) [inline packages/nft-checkout/src/operations/evm/index.ts:359]: why

File: packages/react-nft-checkout/src/components/SwapTokensSelect/SwapTokensSelect.tsx [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+  Select,
+  SelectChangeEvent,
+} from '@mui/material'
+import { TARGET_TOKEN_SYMBOLS, Token } from '@rarimo/nft-checkout'
+import { EthProviderRpcError } from '@rarimo/provider'
+import { Network } from 'iconoir-react'
+import { useEffect, useMemo, useState } from 'react'
+
+import { ErrorText, LoadingIndicator } from '@/components'
+import styles from '@/components/BridgeChainSelect/BridgeChainSelect.module.css'
```
Comment (napalmpapalam) [inline packages/react-nft-checkout/src/components/SwapTokensSelect/SwapTokensSelect.tsx:16]: extract this file to src/styles

File: packages/react-nft-checkout/src/components/SwapTokensSelect/SwapTokensSelect.tsx [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+    selectedSwapToken,
+  } = useDappContext()
+
+  const isDisabled = (token?: Token) => {
+    if (token?.isNative) return true
+    for (const key in TARGET_TOKEN_SYMBOLS) {
+      if (TARGET_TOKEN_SYMBOLS[key] === token?.symbol) return true
+    }
+    return false
+  }
```
Comment (napalmpapalam) [inline packages/react-nft-checkout/src/components/SwapTokensSelect/SwapTokensSelect.tsx:35]:
```ts
  const isDisabled = (token?: Token) => {
    if (token?.isNative) return true
    return Object.values(TARGET_TOKEN_SYMBOLS).some(i => i === token?.symbol)
  }
```

File: packages/react-nft-checkout/src/components/SwapTokensSelect/SwapTokensSelect.tsx [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+
+  const [isLoading, setIsLoading] = useState(true)
+  const [loadingErrorText, setLoadingErrorText] = useState('')
+  const [tokens, setTokens] = useState<Token[]>([])
+
+  useEffect(() => {
+    const fetchSupportedTokens = async () => {
+      try {
+        setIsLoading(true)
+        setLoadingErrorText('')
```
Comment (napalmpapalam) [inline packages/react-nft-checkout/src/components/SwapTokensSelect/SwapTokensSelect.tsx:57]: move above try

File: packages/react-nft-checkout/src/components/SwapTokensSelect/SwapTokensSelect.tsx [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+        const supportedTokens = selectedChain
+          ? (await getSupportedTokens?.(selectedChain)) ?? []
+          : []
+        setTokens(supportedTokens)
+      } catch (error) {
+        setLoadingErrorText(
+          (error as unknown as EthProviderRpcError)?.message ||
+            'An error occurred while loading tokens. Change the network or try again later.',
+        )
```
Comment (napalmpapalam) [inline packages/react-nft-checkout/src/components/SwapTokensSelect/SwapTokensSelect.tsx:66]:
```ts
       setLoadingErrorText(
          (<Error>e)?.message ||
            'An error occurred while loading tokens. Change the network or try again later.',
        )
```

**Review by napalmpapalam — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (WhiteNik16) [reply CHANGELOG.md:14]: fixed

File: packages/react-nft-checkout/src/components/SwapTokensSelect/SwapTokensSelect.tsx (hunk 6 above)
Comment (WhiteNik16) [reply packages/react-nft-checkout/src/components/SwapTokensSelect/SwapTokensSelect.tsx:35]: fixed

File: packages/react-nft-checkout/src/components/SwapTokensSelect/SwapTokensSelect.tsx (hunk 7 above)
Comment (WhiteNik16) [reply packages/react-nft-checkout/src/components/SwapTokensSelect/SwapTokensSelect.tsx:57]: fixed

File: packages/nft-checkout/src/operations/evm/index.ts (hunk 4 above)
Comment (WhiteNik16) [reply packages/nft-checkout/src/operations/evm/index.ts]: fixed

File: packages/react-nft-checkout/src/components/SwapTokensSelect/SwapTokensSelect.tsx (hunk 8 above)
Comment (WhiteNik16) [reply packages/react-nft-checkout/src/components/SwapTokensSelect/SwapTokensSelect.tsx]: fixed

**Review by napalmpapalam — CHANGES_REQUESTED** [review]

File: packages/react-nft-checkout/src/components/BridgeChainSelect/BridgeChainSelect.tsx [hunk 9]
```diff
@@ -12,7 +12,7 @@ import { useMemo } from 'react'
 
 import { useDappContext } from '@/hooks'
 
-import styles from './BridgeChainSelect.module.css'
+import styles from '../../styles/BridgeChainSelect.module.css'
```
[the remark this answers is not in the document]
Comment (WhiteNik16) [reply packages/react-nft-checkout/src/components/BridgeChainSelect/BridgeChainSelect.tsx]: fixed

**Review by napalmpapalam — APPROVED** [review]

**Pull request merged**